transitions/select.h

```diff
@@ -0,0 +1,47 @@
+#ifndef TRANSITIONS_SELECT_H
+#define TRANSITIONS_SELECT_H
+
+#include "random.h"
+
+// Usage: TrSelect<SELECTION, TR1, TR2, ...>
+// SELECTION: FUNCTION
+// TR1, TR2: TRANSITION
+// return value: TRANSITION
+// transition option is picked from the specified list of
+// transitions based on Int<>
+// with Int<0> representing first transition
+
+template<class F, class... TRANSITION>
+class TrSelect {
+public:
+  void begin() {
+    begin_ = true;
+  }
+  void run(BladeBase* blade) {
+    f_.run(blade);
+    if (begin_) {
+      begin_ = false;
+      unsigned int n = f_.getInteger(0);
+      selected_ = transitions_.get(n % sizeof...(TRANSITION));
+      selected_->begin();
+    }
+      selected_->run(blade); 
```

Comment:
This and the next line looks indented wrong.
Might just be github, if so I'll fix it afterwards.

Reply:
it was out of place, should be fixed